---
title: "A/B_compartments"
author: "Vilde Gustafsson Bagstevold"
date: "2025-11-20"
output: html_document
---

# Packages

```{r setup, include=FALSE}
knitr::opts_chunk$set(message = FALSE, warning = FALSE)

library(tidyverse)
library(readr)
library(dplyr)
library(tidyr)
library(GenomicRanges)
library(GenomeInfoDb)

# color palette

```

```{r, include=FALSE}
### import Ensemble genes and repeats
mapping <- read_tsv("Ssal_v3.1_chromosomes.tsv") 
gene <- read_tsv("https://salmobase.org/datafiles/TSV/genes/AtlanticSalmon/Ssal_v3.1/Ensembl_genes.tsv")
repeats <- read_tsv("https://salmobase.org/datafiles/datasets/Salmon_repeats/Ssal_v3.1_repeats_5.2.bed", col_names = F)
names(repeats) <- c("chromosome", "start", "end", "annotation")

### loading table with all genes,ssal included
data <- read.delim("all_eve_genes-kopi.txt", header = TRUE)
```

# Loading HiC files

```{r}

### loading HiC files, gill and liver CHOOSING THE RIGHT E123
hicG <- read.delim("HiC_G.250000_compartments.cis.vecs.tsv", header = TRUE, sep = "\t")
hicL <- read.delim("HiC_L.250000_compartments.cis.vecs.tsv", header = TRUE, sep = "\t")

#to be done when correlation test is done:
#selected_columns <- c("chrom", "start", "end", "weight", "E1")
#filtered_data <- data[, selected_columns]


compartments <- inner_join(
  hicG, hicL, by = c("chrom", "start", "end")
) %>%
  dplyr::rename(
    weight.G = weight.x,
    weight.L = weight.y,
    E1_gill  = E1.x,
    E1_liver = E1.y,
    E2_gill  = E2.x,
    E2_liver = E2.y,
    E3_gill  = E3.x,
    E3_liver = E3.y
  )

```

# Perform correlation test between GC-content and eigenvectors

```{r}
# Load GC content file
gc <- read.table("genome_gc.txt", header = TRUE)

# Merge by common columns 
gc_compartments <- merge(gc, compartments, by = c("chrom", "start"))

library(rtracklayer)
library(GenomicRanges)

# Import GFF3
gff <- import("Salmo_salar.Ssal_v3.1.106.gff3")

# Filter for genes
onlygenes <- gff[gff$type == "gene"]

# Define bins based on gc_compartments
bins <- GRanges(
  seqnames = gc_compartments$chrom,
  ranges = IRanges(start = gc_compartments$start, end = gc_compartments$end)
)

library(GenomeInfoDb)
seqlevelsStyle(onlygenes) <- seqlevelsStyle(bins)


# Count genes overlapping each bin
gene_counts <- countOverlaps(bins, onlygenes)

# Add raw gene counts to gc_compartments
gc_compartments$gene_density <- gene_counts

# Calculate normalized gene density by bin size
bin_size <- gc_compartments$end.x - gc_compartments$start
gc_compartments$gene_density_norm <- gene_counts / bin_size

gene_counts <- countOverlaps(bins, onlygenes)
gc_compartments$gene_density <- gene_counts
gc_compartments$gene_density_norm <- gene_counts / (gc_compartments$end.x - gc_compartments$start)

var(gc_compartments$gene_density_norm)


# Correlation of GC vs eigenvectors
gc_cor_p <- cor(gc_compartments[, c("GC", "E1_liver", "E2_liver", "E3_liver", "E1_gill", "E2_gill", "E3_gill")],
              method = "pearson", use = "complete.obs")
gc_cor_s <- cor(gc_compartments[, c("GC", "E1_liver", "E2_liver", "E3_liver", "E1_gill", "E2_gill", "E3_gill")],
              method = "spearman", use = "complete.obs")

# Correlation of gene density vs eigenvectors
gene_cor_p <- cor(gc_compartments[, c("gene_density_norm", "E1_liver", "E2_liver", "E3_liver", "E1_gill", "E2_gill", "E3_gill")],
                method = "pearson", use = "complete.obs")

gene_cor_s <- cor(gc_compartments[, c("gene_density_norm", "E1_liver", "E2_liver", "E3_liver", "E1_gill", "E2_gill", "E3_gill")],
                method = "spearman", use = "complete.obs")


print(gc_cor_s["GC", ])
print(gc_cor_p["GC", ])
print(gene_cor_p["gene_density_norm", ])
print(gene_cor_s["gene_density_norm", ])

```

```{r}

# Add the 'Compartment' column based on the condition, and add switch status
#### FIGURE OUT EIGENVECTOR
compartments <- compartments %>%
  mutate(
    comp_gill  = ifelse(E1_gill  > 0, "A", "B"),
    comp_liver = ifelse(E1_liver > 0, "A", "B"),
    switch_status = case_when(
      comp_gill == "A" & comp_liver == "A" ~ "A→A (stable)",
      comp_gill == "B" & comp_liver == "B" ~ "B→B (stable)",
      comp_gill == "A" & comp_liver == "B" ~ "A→B (switch)",
      comp_gill == "B" & comp_liver == "A" ~ "B→A (switch)"
    )
  )
```

# Making a table with compartments and gene annotations
 
```{r}
### load all input files +++


### prepare mapping table, keeping only chromosome rows
mapping_chr <- mapping %>%
  filter(is_chromosome == TRUE) %>%
  mutate(chrom = paste0("chr", name)) %>%
  select(seqname, chrom)

### merge gene with mapping to get chromosome names

gene <- gene %>%
  left_join(mapping_chr, by = "seqname") %>%
  select(chrom, start, end, gene_id)

### merge data with gene position
data_mapped <- data %>%
  left_join(gene, by = c("Ssal" = "gene_id")) %>%
  filter(!is.na(chrom))  # remove genes on contigs


### merge with compartments based on overlap

merged <- data_mapped %>%
  rowwise() %>%
  mutate(
    match_idx = which(compartments$chrom == chrom &
                        compartments$start <= start &
                        compartments$end >= end)[1],
    E1_gill = ifelse(!is.na(match_idx), compartments$E1_gill[match_idx], NA),
    E2_gill = ifelse(!is.na(match_idx), compartments$E2_gill[match_idx], NA),
    E3_gill = ifelse(!is.na(match_idx), compartments$E3_gill[match_idx], NA),
    E1_liver = ifelse(!is.na(match_idx), compartments$E1_liver[match_idx], NA),
    E2_liver = ifelse(!is.na(match_idx), compartments$E2_liver[match_idx], NA),
    E3_liver = ifelse(!is.na(match_idx), compartments$E3_liver[match_idx], NA)
  ) %>%
  ungroup() %>%
  select(-match_idx)

### save the combined file
write_tsv(merged, "combined.tsv")
```

```{r}

```

